### PR TITLE
Refactor is_aws, is_azure, is_gcp to Component

### PR DESCRIPTION
**Pull request by xiangce**
Signed-off-by: Xiangce Liu <xiangceliu@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:

The `is_aws`, `is_azure`, `is_gcp` look more like Components than datasources, this PR move them to the `insights.components.cloud_provider` module.

### CODE REVIEW

**Review by bfahr — APPROVED** [review]

File: insights/components/cloud_provider.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+executing.
+
+"""
+from insights.core.dr import SkipComponent
+from insights.core.plugins import component
+from insights.combiners.cloud_provider import CloudProvider
+
+
+@component(CloudProvider)
+class IsAWS(object):
```
Comment (bfahr) [inline insights/components/cloud_provider.py:18]: @xiangce you'll need to add these to the default manifest in `insights/collect.py` in order for these components to load during collection.  When you look there you'll see how the IsRhel components were added.

**Review by bfahr — CHANGES_REQUESTED** [review]
Premature approval.  Just one change and everything else looks OK.

File: insights/components/cloud_provider.py (hunk 1 above)
Comment (xiangce) [reply insights/components/cloud_provider.py:18]: Added. @bfahr Thanks!

**Review by bfahr — APPROVED** [review]

**Comment by xiangce** [comment]
Relate to #3087

**Pull request merged**